lib/matplotlib/axes/_axes.py

```diff
@@ -7293,61 +7293,82 @@ def specgram(self, x, NFFT=None, Fs=None, Fc=None, detrend=None,
-            corner of the axes.
+            corner of the axes. The convention 'upper' is typically used for
+            matrices and images.
+            If not given, :rc:`image.origin` is used, defaulting to 'upper'.
 
-        aspect : ['auto' | 'equal' | scalar], optional, default: "equal"
 
-            If 'equal', and `extent` is None, changes the axes aspect ratio to
-            match that of the image. If `extent` is not `None`, the axes
-            aspect ratio is changed to match that of the extent.
+        aspect : {'equal', 'auto', None} or float, optional
+            Controls the aspect ratio of the axes. The aspect is of particular
+            relevance for images since it may distort the image, i.e. pixel
+            will not be square.
 
+            This parameter is a shortcut for explicitly calling
+            `.Axes.set_aspect`. See there for further details.
 
-            If 'auto', changes the image aspect ratio to match that of the
-            axes.
+            - 'equal': Ensures an aspect ratio of 1. Pixels will be square
+              (unless pixel sizes are explicitly made non-square in data
+              coordinates using *extent*).
+            - 'auto': The axes is kept fixed and the aspect is adjusted so
+              that the data fit in the axes. In general, this will result in
+              non-square pixels.
+            - *None*: Use :rc:`image.aspect` (default: 'equal').
 
-            If None, default to rc ``image.aspect`` value.
+            Default: 'equal'
 
-        Two plotting styles are available: image or marker. Both
-        are available for full arrays, but only the marker style
-        works for :class:`scipy.sparse.spmatrix` instances.
+        Returns
+        -------
+        ret : `~matplotlib.image.AxesImage` or `.Line2D`
+            The return type depends on the plotting style (see above).
 
-        If *marker* and *markersize* are *None*, an image will be
-        returned and any remaining kwargs are passed to
-        :func:`~matplotlib.pyplot.imshow`; else, a
-        :class:`~matplotlib.lines.Line2D` object will be returned with
-        the value of marker determining the marker type, and any
-        remaining kwargs passed to the
-        :meth:`~matplotlib.axes.Axes.plot` method.
+        Additional Parameters
+        ---------------------
+        **kwargs
+            The supported additional parameters depend on the plotting style.
 
-        If *marker* and *markersize* are *None*, useful kwargs include:
+            For the image style, you can pass any parameters supported by
+            `~.Axes.imshow`. Commonly used parameters include *cmap* and
+            *alpha*.
 
-        * *cmap*
-        * *alpha*
+            For the marker style, you can pass any parameters
+            supported by `~.Axes.plot. Commonly used parameters include
+            *color*, *marker* and *markersize*.
```

Comment:
You cannot pass `linestyle`.

Also *marker* and *markersize* are already arguments of `spy`, so one wouldn't list them as "Additional Parameters" as well,right?